sympy/integrals/tests/test_integrals.py

```diff
@@ -1649,6 +1649,20 @@ def test_issue_2975():
     y = Symbol('y')
     assert integrate(1/(y**2+C)**(S(3)/2), (y, -w/2, w/2)) == w/(C**(S(3)/2)*sqrt(1 + w**2/(4*C)))
 
+def test_issue_7827():
+    x, n, N = symbols('x n N')
+    assert integrate(summation(x*n, (n, 1, N)), x) == x**2*(N**2/4 + N/4)
```

Comment:
This doesn't look correct. If `N = 2`, then we have `integrate(x + x**2, x) == x**2/2 + x**3/3` but the right hand side is `x**2*(1 + 1/2)`.

Reply:
@jksuom actually when ```N = 2``` the expression ```summation(x*n, (n, 1, N))``` becomes ```x*1 + x*2```(= ```x*3```) and integration of ```x*3``` = ```x**2 * (3/2)``` which is ```x**2*(2**2/4 + 2/4)``` = ```x**2*3/2```

Reply:
Sorry, my mistake. What would be the result with `x**n`?

Reply:
@jksuom For ```x**n``` the output is:
```-N*x**2*x**N*lerchphi(x, 1, N + 2)*gamma(N + 2)/gamma(N + 3) - 2*x**2*x**N*lerchphi(x, 1, N + 2)*gamma(N + 2)/gamma(N + 3) - x - log(x - 1)```

I think the reason for this complicated output maybe because the summation takes place first and then the expanded expression is integrated. So, when we pass ```summation(x**n, (n, 1, N))``` this summation gets evaluated first and then it is integrated to give the complicated output we see above.

Reply:
I think that the result could be `Sum(x**(n+1)/(n+1), (n, 1, N))`.

Reply:
Ya agreed. I would try some changes and see if I can get this result.

Reply:
@jksuom I think when we use something like ```summation```, this function expands first and then integrates. But when I tried using ```integrate(Sum(x**n, (n, 1, N)), x)``` to integrate instead, the output was ```Sum(Piecewise((x**(n + 1)/(n + 1), Ne(n, -1)), (log(x), True)), (n, 1, N))```.